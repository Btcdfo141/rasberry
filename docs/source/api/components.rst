:mod:`spencerassistant.components`
===============================

air\_quality
--------------------------------------------

.. automodule:: spencerassistant.components.air_quality
    :members:
    :undoc-members:
    :show-inheritance:

alarm\_control\_panel
--------------------------------------------

.. automodule:: spencerassistant.components.alarm_control_panel
    :members:
    :undoc-members:
    :show-inheritance:

binary\_sensor
--------------------------------------------

.. automodule:: spencerassistant.components.binary_sensor
    :members:
    :undoc-members:
    :show-inheritance:

camera
---------------------------

.. automodule:: spencerassistant.components.camera
    :members:
    :undoc-members:
    :show-inheritance:

calendar
---------------------------

.. automodule:: spencerassistant.components.calendar
    :members:
    :undoc-members:
    :show-inheritance:

climate
---------------------------

.. automodule:: spencerassistant.components.climate
    :members:
    :undoc-members:
    :show-inheritance:

conversation
---------------------------

.. automodule:: spencerassistant.components.conversation
    :members:
    :undoc-members:
    :show-inheritance:

cover
---------------------------

.. automodule:: spencerassistant.components.cover
    :members:
    :undoc-members:
    :show-inheritance:

device\_tracker
---------------------------

.. automodule:: spencerassistant.components.device_tracker
    :members:
    :undoc-members:
    :show-inheritance:

fan
---------------------------

.. automodule:: spencerassistant.components.fan
    :members:
    :undoc-members:
    :show-inheritance:

light
---------------------------

.. automodule:: spencerassistant.components.light
    :members:
    :undoc-members:
    :show-inheritance:

lock
---------------------------

.. automodule:: spencerassistant.components.lock
    :members:
    :undoc-members:
    :show-inheritance:

media\_player
---------------------------

.. automodule:: spencerassistant.components.media_player
    :members:
    :undoc-members:
    :show-inheritance:

notify
---------------------------

.. automodule:: spencerassistant.components.notify
    :members:
    :undoc-members:
    :show-inheritance:

remote
---------------------------

.. automodule:: spencerassistant.components.remote
    :members:
    :undoc-members:
    :show-inheritance:

switch
---------------------------

.. automodule:: spencerassistant.components.switch
    :members:
    :undoc-members:
    :show-inheritance:

sensor
-------------------------------------

.. automodule:: spencerassistant.components.sensor
    :members:
    :undoc-members:
    :show-inheritance:

vacuum
-------------------------------------

.. automodule:: spencerassistant.components.vacuum
    :members:
    :undoc-members:
    :show-inheritance:

water\_heater
-------------------------------------

.. automodule:: spencerassistant.components.water_heater
    :members:
    :undoc-members:
    :show-inheritance:

weather
---------------------------

.. automodule:: spencerassistant.components.weather
    :members:
    :undoc-members:
    :show-inheritance:

webhook
---------------------------

.. automodule:: spencerassistant.components.webhook
    :members:
    :undoc-members:
    :show-inheritance:
