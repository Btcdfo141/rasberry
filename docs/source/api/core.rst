.. _core_module:

:mod:`spencerassistant.core`
-------------------------

.. automodule:: spencerassistant.core
    :members: