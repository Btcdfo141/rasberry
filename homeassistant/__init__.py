"""Init file for Home Assistant."""


test2
