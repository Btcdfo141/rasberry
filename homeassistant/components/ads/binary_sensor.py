"""Support for ADS binary sensors."""
from homeassistant.components.binary_sensor import (
    DEVICE_CLASS_MOVING,
    BinarySensorEntity,
)
from homeassistant.const import CONF_BINARY_SENSORS, CONF_DEVICE_CLASS, CONF_NAME

from . import CONF_ADS_VAR, DATA_ADS, STATE_KEY_STATE, AdsEntity


def setup_platform(hass, config, add_entities, discovery_info=None):
    """Set up the Binary Sensor platform for ADS."""
    entities = []

    if discovery_info is None:  # pragma: no cover
        return

    ads_hub = hass.data.get(DATA_ADS)

    for entry in discovery_info[CONF_BINARY_SENSORS]:
        ads_var = entry.get(CONF_ADS_VAR)
        name = entry.get(CONF_NAME)
        device_class = entry.get(CONF_DEVICE_CLASS)
        entities.append(AdsBinarySensor(ads_hub, name, ads_var, device_class))

    add_entities(entities)


class AdsBinarySensor(AdsEntity, BinarySensorEntity):
    """Representation of ADS binary sensors."""

    def __init__(self, ads_hub, name, ads_var, device_class):
        """Initialize ADS binary sensor."""
        super().__init__(ads_hub, name, ads_var)
        self._attr_device_class = device_class or DEVICE_CLASS_MOVING

    async def async_added_to_hass(self):
        """Register device notification."""
        await self.async_initialize_device(self._ads_var, self._ads_hub.PLCTYPE_BOOL)

    @property
    def is_on(self) -> bool:
        """Return True if the entity is on."""
        return self._state_dict[STATE_KEY_STATE]
