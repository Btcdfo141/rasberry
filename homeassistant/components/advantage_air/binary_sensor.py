"""Binary Sensor platform for Advantage Air integration."""
from homeassistant.components.binary_sensor import (
    DEVICE_CLASS_MOTION,
    DEVICE_CLASS_PROBLEM,
    BinarySensorEntity,
)
from homeassistant.const import CONF_IP_ADDRESS

from .const import DOMAIN


async def async_setup_platform(hass, config, async_add_entities, discovery_info=None):
    """Platform setup isnt required."""
    return True


async def async_setup_entry(hass, config_entry, async_add_entities):
    """Set up AdvantageAir motion platform."""

    instance = hass.data[DOMAIN][config_entry.data[CONF_IP_ADDRESS]]

    if "aircons" in instance["coordinator"].data:
        entities = []
        for _, ac_index in enumerate(instance["coordinator"].data["aircons"]):
            entities.append(AdvantageAirZoneFilter(instance, ac_index))
            for _, zone_index in enumerate(
                instance["coordinator"].data["aircons"][ac_index]["zones"]
            ):
                # Only add motion sensor when motion is enabled
                if (
                    instance["coordinator"].data["aircons"][ac_index]["zones"][
                        zone_index
                    ]["motionConfig"]
                    == 0
                ):
                    entities.append(
                        AdvantageAirZoneMotion(instance, ac_index, zone_index)
                    )
        async_add_entities(entities)
    return True


class AdvantageAirZoneFilter(BinarySensorEntity):
    """AdvantageAir Filter."""

    def __init__(self, instance, ac_index):
        """Initialize the Advantage Air Zone Filter."""
        self.coordinator = instance["coordinator"]
        self.async_change = instance["async_change"]
        self.device = instance["device"]
        self.ac_index = ac_index

    @property
    def name(self):
        """Return the name."""
        return (
            f"{self.coordinator.data['aircons'][self.ac_index]['info']['name']} Filter"
        )

    @property
    def unique_id(self):
        """Return a unique id."""
        return f"{self.coordinator.data['system']['rid']}-{self.ac_index}-binary:filter"

    @property
    def device_class(self):
        """Return the device class of the vent."""
        return DEVICE_CLASS_PROBLEM

    @property
    def is_on(self):
        """Return if filter needs cleaning."""
        return self.coordinator.data["aircons"][self.ac_index]["info"][
            "filterCleanStatus"
        ]

    @property
    def should_poll(self):
        """No polling needed."""
        return False

    @property
    def available(self):
        """Return if platform is avaliable."""
        return self.coordinator.last_update_success

    @property
    def device_info(self):
        """Return parent device information."""
        return self.device

    async def async_added_to_hass(self):
        """When entity is added to hass."""
        self.async_on_remove(
            self.coordinator.async_add_listener(self.async_write_ha_state)
        )

    async def async_update(self):
        """Request update."""
        await self.coordinator.async_request_refresh()


class AdvantageAirZoneMotion(BinarySensorEntity):
    """AdvantageAir Zone Motion."""

    def __init__(self, instance, ac_index, zone_index):
        """Initialize the Advantage Air Zone Motion sensor."""
        self.coordinator = instance["coordinator"]
        self.async_change = instance["async_change"]
        self.device = instance["device"]
        self.ac_index = ac_index
        self.zone_index = zone_index

    @property
    def name(self):
        """Return the name."""
        return f"{self.coordinator.data['aircons'][self.ac_index]['zones'][self.zone_index]['name']} Motion"

    @property
    def unique_id(self):
        """Return a unique id."""
        return f"{self.coordinator.data['system']['rid']}-{self.ac_index}-{self.zone_index}-binary:motion"

    @property
    def device_class(self):
        """Return the device class of the vent."""
        return DEVICE_CLASS_MOTION

    @property
    def is_on(self):
        """Return if motion is detect."""
        return self.coordinator.data["aircons"][self.ac_index]["zones"][
            self.zone_index
        ]["motion"]

    @property
    def device_state_attributes(self):
        """Return additional motion configuration."""
        return {
            "motionConfig": self.coordinator.data["aircons"][self.ac_index]["zones"][
                self.zone_index
            ]["motionConfig"]
        }

    @property
    def should_poll(self):
        """No polling needed."""
        return False

    @property
    def available(self):
        """Return if platform is avaliable."""
        return self.coordinator.last_update_success

    @property
    def device_info(self):
        """Return parent device information."""
        return self.device

    async def async_added_to_hass(self):
        """When entity is added to hass."""
        self.async_on_remove(
            self.coordinator.async_add_listener(self.async_write_ha_state)
        )

    async def async_update(self):
        """Request update."""
        await self.coordinator.async_request_refresh()
