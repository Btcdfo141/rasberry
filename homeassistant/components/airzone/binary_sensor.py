"""Support for the Airzone sensors."""
from __future__ import annotations

from collections.abc import Mapping
from typing import Any, Final

from aioairzone.const import (
    AZD_AIR_DEMAND,
    AZD_DEMAND,
    AZD_ERRORS,
    AZD_FLOOR_DEMAND,
    AZD_NAME,
    AZD_PROBLEMS,
    AZD_ZONES,
)

from homeassistant.components.binary_sensor import (
    DEVICE_CLASS_PROBLEM,
    DEVICE_CLASS_RUNNING,
    BinarySensorEntity,
    BinarySensorEntityDescription,
)
from homeassistant.config_entries import ConfigEntry
from homeassistant.core import HomeAssistant
from homeassistant.helpers.entity_platform import AddEntitiesCallback

from . import AirzoneEntity
from .const import DOMAIN
from .coordinator import AirzoneUpdateCoordinator

BINARY_SENSOR_TYPES: Final[tuple[BinarySensorEntityDescription, ...]] = (
    BinarySensorEntityDescription(
        device_class=DEVICE_CLASS_RUNNING,
        key=AZD_DEMAND,
        name="Demand",
    ),
    BinarySensorEntityDescription(
        device_class=DEVICE_CLASS_PROBLEM,
        key=AZD_PROBLEMS,
        name="Problem",
    ),
)


async def async_setup_entry(
    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
) -> None:
    """Add Airzone binary sensors from a config_entry."""
    coordinator = hass.data[DOMAIN][entry.entry_id]

    binary_sensors = []
    for system_zone_id, zone_data in coordinator.data[AZD_ZONES].items():
        for description in BINARY_SENSOR_TYPES:
            if description.key in zone_data:
                binary_sensors.append(
                    AirzoneBinarySensor(
                        coordinator,
                        description,
                        entry,
                        system_zone_id,
                        zone_data,
                    )
                )

    async_add_entities(binary_sensors)


class AirzoneBinarySensor(AirzoneEntity, BinarySensorEntity):
    """Define an Airzone sensor."""

    def __init__(
        self,
        coordinator: AirzoneUpdateCoordinator,
        description: BinarySensorEntityDescription,
        entry: ConfigEntry,
        system_zone_id: str,
        zone_data: dict[str, Any],
    ) -> None:
        """Initialize."""
        super().__init__(coordinator, entry, system_zone_id, zone_data)
        self._attr_name = f"{zone_data[AZD_NAME]} {description.name}"
        self._attr_unique_id = f"{entry.entry_id}_{system_zone_id}_{description.key}"
        self.entity_description = description

    @property
    def extra_state_attributes(self) -> Mapping[str, Any] | None:
        """Return state attributes."""
        _state_attr = None
        if self.entity_description.key == AZD_DEMAND:
            _state_attr = {
                "air": self.get_zone_value(AZD_AIR_DEMAND),
                "floor": self.get_zone_value(AZD_FLOOR_DEMAND),
            }
        elif self.entity_description.key == AZD_PROBLEMS:
            _state_attr = {
                "errors": self.get_zone_value(AZD_ERRORS),
            }
        return _state_attr

    @property
    def is_on(self) -> bool | None:
        """Return true if the binary sensor is on."""
        return self.get_zone_value(self.entity_description.key)
