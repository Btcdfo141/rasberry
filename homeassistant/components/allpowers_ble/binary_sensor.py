"""Allpowers BLE integration binary sensor platform."""


from allpowers_ble import AllpowersBLE

from homeassistant.components.binary_sensor import (
    BinarySensorDeviceClass,
    BinarySensorEntity,
    BinarySensorEntityDescription,
)
from homeassistant.config_entries import ConfigEntry
from homeassistant.core import HomeAssistant, callback
from homeassistant.helpers import device_registry as dr
from homeassistant.helpers.entity_platform import AddEntitiesCallback
from homeassistant.helpers.update_coordinator import CoordinatorEntity

from .const import DOMAIN
from .coordinator import AllpowersBLECoordinator
from .models import AllpowersBLEData

ENTITY_DESCRIPTIONS = [
    BinarySensorEntityDescription(
        key="is_moving",
        device_class=BinarySensorDeviceClass.MOTION,
        has_entity_name=True,
        name="Motion",
    ),
    BinarySensorEntityDescription(
        key="is_static",
        device_class=BinarySensorDeviceClass.OCCUPANCY,
        has_entity_name=True,
        name="Occupancy",
    ),
]


async def async_setup_entry(
    hass: HomeAssistant,
    entry: ConfigEntry,
    async_add_entities: AddEntitiesCallback,
) -> None:
    """Set up the platform for AllpowersBLE."""
    data: AllpowersBLEData = hass.data[DOMAIN][entry.entry_id]
    async_add_entities(
        AllpowersBLEBinarySensor(
            data.coordinator, data.device, entry.title, description
        )
        for description in ENTITY_DESCRIPTIONS
    )


class AllpowersBLEBinarySensor(CoordinatorEntity, BinarySensorEntity):
    """Moving/static sensor for AllpowersBLE."""

    def __init__(
        self,
        coordinator: AllpowersBLECoordinator,
        device: AllpowersBLE,
        name: str,
        description: BinarySensorEntityDescription,
    ) -> None:
        """Initialize the sensor."""
        super().__init__(coordinator)
        self._coordinator = coordinator
        self._key = description.key
        self._device = device
        self.entity_description = description
        self._attr_unique_id = f"{device.address}_{self._key}"
        self._attr_device_info = dr.DeviceInfo(
            name=name,
            connections={(dr.CONNECTION_BLUETOOTH, device.address)},
        )
        self._attr_is_on = getattr(self._device, self._key)

    @callback
    def _handle_coordinator_update(self) -> None:
        """Handle updated data from the coordinator."""
        self._attr_is_on = getattr(self._device, self._key)
        self.async_write_ha_state()

    @property
    def available(self) -> bool:
        """Unavailable if coordinator isn't connected."""
        return self._coordinator.connected and super().available
