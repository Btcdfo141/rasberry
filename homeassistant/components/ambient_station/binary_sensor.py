"""Support for Ambient Weather Station binary sensors."""
from __future__ import annotations

from homeassistant.components.binary_sensor import (
    DEVICE_CLASS_BATTERY,
    DEVICE_CLASS_CONNECTIVITY,
    BinarySensorEntity,
    BinarySensorEntityDescription,
)
from homeassistant.config_entries import ConfigEntry
from homeassistant.const import ATTR_NAME
from homeassistant.core import HomeAssistant, callback
from homeassistant.helpers.entity_platform import AddEntitiesCallback

from . import AmbientWeatherEntity
from .const import (
    ATTR_LAST_DATA,
    DATA_CLIENT,
    DOMAIN,
    TYPE_BATT1,
    TYPE_BATT2,
    TYPE_BATT3,
    TYPE_BATT4,
    TYPE_BATT5,
    TYPE_BATT6,
    TYPE_BATT7,
    TYPE_BATT8,
    TYPE_BATT9,
    TYPE_BATT10,
    TYPE_BATT_CO2,
    TYPE_BATTOUT,
    TYPE_PM25_BATT,
    TYPE_PM25IN_BATT,
    TYPE_RELAY1,
    TYPE_RELAY2,
    TYPE_RELAY3,
    TYPE_RELAY4,
    TYPE_RELAY5,
    TYPE_RELAY6,
    TYPE_RELAY7,
    TYPE_RELAY8,
    TYPE_RELAY9,
    TYPE_RELAY10,
)

BINARY_SENSOR_DESCRIPTIONS = (
    BinarySensorEntityDescription(
        key=TYPE_BATTOUT,
        name="Battery",
        device_class=DEVICE_CLASS_BATTERY,
    ),
    BinarySensorEntityDescription(
        key=TYPE_BATT1,
        name="Battery 1",
        device_class=DEVICE_CLASS_BATTERY,
    ),
    BinarySensorEntityDescription(
        key=TYPE_BATT2,
        name="Battery 2",
        device_class=DEVICE_CLASS_BATTERY,
    ),
    BinarySensorEntityDescription(
        key=TYPE_BATT3,
        name="Battery 3",
        device_class=DEVICE_CLASS_BATTERY,
    ),
    BinarySensorEntityDescription(
        key=TYPE_BATT4,
        name="Battery 4",
        device_class=DEVICE_CLASS_BATTERY,
    ),
    BinarySensorEntityDescription(
        key=TYPE_BATT5,
        name="Battery 5",
        device_class=DEVICE_CLASS_BATTERY,
    ),
    BinarySensorEntityDescription(
        key=TYPE_BATT6,
        name="Battery 6",
        device_class=DEVICE_CLASS_BATTERY,
    ),
    BinarySensorEntityDescription(
        key=TYPE_BATT7,
        name="Battery 7",
        device_class=DEVICE_CLASS_BATTERY,
    ),
    BinarySensorEntityDescription(
        key=TYPE_BATT8,
        name="Battery 8",
        device_class=DEVICE_CLASS_BATTERY,
    ),
    BinarySensorEntityDescription(
        key=TYPE_BATT9,
        name="Battery 9",
        device_class=DEVICE_CLASS_BATTERY,
    ),
    BinarySensorEntityDescription(
        key=TYPE_BATT10,
        name="Battery 10",
        device_class=DEVICE_CLASS_BATTERY,
    ),
    BinarySensorEntityDescription(
        key=TYPE_BATT_CO2,
        name="CO2 Battery",
        device_class=DEVICE_CLASS_BATTERY,
    ),
    BinarySensorEntityDescription(
        key=TYPE_PM25IN_BATT,
        name="PM25 Indoor Battery",
        device_class=DEVICE_CLASS_BATTERY,
    ),
    BinarySensorEntityDescription(
        key=TYPE_PM25_BATT,
        name="PM25 Battery",
        device_class=DEVICE_CLASS_BATTERY,
    ),
    BinarySensorEntityDescription(
        key=TYPE_RELAY1,
        name="Relay 1",
        device_class=DEVICE_CLASS_CONNECTIVITY,
    ),
    BinarySensorEntityDescription(
        key=TYPE_RELAY2,
        name="Relay 2",
        device_class=DEVICE_CLASS_CONNECTIVITY,
    ),
    BinarySensorEntityDescription(
        key=TYPE_RELAY3,
        name="Relay 3",
        device_class=DEVICE_CLASS_CONNECTIVITY,
    ),
    BinarySensorEntityDescription(
        key=TYPE_RELAY4,
        name="Relay 4",
        device_class=DEVICE_CLASS_CONNECTIVITY,
    ),
    BinarySensorEntityDescription(
        key=TYPE_RELAY5,
        name="Relay 5",
        device_class=DEVICE_CLASS_CONNECTIVITY,
    ),
    BinarySensorEntityDescription(
        key=TYPE_RELAY6,
        name="Relay 6",
        device_class=DEVICE_CLASS_CONNECTIVITY,
    ),
    BinarySensorEntityDescription(
        key=TYPE_RELAY7,
        name="Relay 7",
        device_class=DEVICE_CLASS_CONNECTIVITY,
    ),
    BinarySensorEntityDescription(
        key=TYPE_RELAY8,
        name="Relay 8",
        device_class=DEVICE_CLASS_CONNECTIVITY,
    ),
    BinarySensorEntityDescription(
        key=TYPE_RELAY9,
        name="Relay 9",
        device_class=DEVICE_CLASS_CONNECTIVITY,
    ),
    BinarySensorEntityDescription(
        key=TYPE_RELAY10,
        name="Relay 10",
        device_class=DEVICE_CLASS_CONNECTIVITY,
    ),
)


async def async_setup_entry(
    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
) -> None:
    """Set up Ambient PWS binary sensors based on a config entry."""
    ambient = hass.data[DOMAIN][DATA_CLIENT][entry.entry_id]

    async_add_entities(
        [
            AmbientWeatherBinarySensor(
                ambient, mac_address, station[ATTR_NAME], description
            )
            for description in BINARY_SENSOR_DESCRIPTIONS
            for mac_address, station in ambient.stations.items()
            if description.key in station[ATTR_LAST_DATA]
        ]
    )


class AmbientWeatherBinarySensor(AmbientWeatherEntity, BinarySensorEntity):
    """Define an Ambient binary sensor."""

    @callback
    def update_from_latest_data(self) -> None:
        """Fetch new state data for the entity."""
        state = self._ambient.stations[self._mac_address][ATTR_LAST_DATA][
            self.entity_description.key
        ]

        if self.entity_description.key in (
            TYPE_BATT1,
            TYPE_BATT10,
            TYPE_BATT2,
            TYPE_BATT3,
            TYPE_BATT4,
            TYPE_BATT5,
            TYPE_BATT6,
            TYPE_BATT7,
            TYPE_BATT8,
            TYPE_BATT9,
            TYPE_BATT_CO2,
            TYPE_BATTOUT,
            TYPE_PM25_BATT,
            TYPE_PM25IN_BATT,
        ):
            self._attr_is_on = state == 0
        else:
            self._attr_is_on = state == 1
