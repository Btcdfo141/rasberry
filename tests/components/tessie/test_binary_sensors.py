"""Test the Tessie binary sensor platform."""
from syrupy import SnapshotAssertion

from homeassistant.components.binary_sensor import DOMAIN as BINARY_SENSOR_DOMAIN
from homeassistant.core import HomeAssistant

from .common import setup_platform


async def test_binary_sensors(hass: HomeAssistant, snapshot: SnapshotAssertion) -> None:
    """Tests that the binary sensor entities are correct."""

    assert len(hass.states.async_all(BINARY_SENSOR_DOMAIN)) == 0

    await setup_platform(hass)

    assert hass.states.async_all(BINARY_SENSOR_DOMAIN) == snapshot(name="all")
