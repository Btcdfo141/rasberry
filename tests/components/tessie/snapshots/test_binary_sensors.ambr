# serializer version: 1
# name: test_binary_sensors[all]
  list([
    StateSnapshot({
      'attributes': ReadOnlyDict({
        'device_class': 'connectivity',
        'friendly_name': 'Test Status',
      }),
      'context': <ANY>,
      'entity_id': 'binary_sensor.test_status',
      'last_changed': <ANY>,
      'last_updated': <ANY>,
      'state': 'on',
    }),
    StateSnapshot({
      'attributes': ReadOnlyDict({
        'device_class': 'heat',
        'friendly_name': 'Test Battery heater',
      }),
      'context': <ANY>,
      'entity_id': 'binary_sensor.test_battery_heater',
      'last_changed': <ANY>,
      'last_updated': <ANY>,
      'state': 'off',
    }),
    StateSnapshot({
      'attributes': ReadOnlyDict({
        'device_class': 'battery_charging',
        'friendly_name': 'Test Charging',
      }),
      'context': <ANY>,
      'entity_id': 'binary_sensor.test_charging',
      'last_changed': <ANY>,
      'last_updated': <ANY>,
      'state': 'on',
    }),
    StateSnapshot({
      'attributes': ReadOnlyDict({
        'friendly_name': 'Test Preconditioning enabled',
      }),
      'context': <ANY>,
      'entity_id': 'binary_sensor.test_preconditioning_enabled',
      'last_changed': <ANY>,
      'last_updated': <ANY>,
      'state': 'off',
    }),
    StateSnapshot({
      'attributes': ReadOnlyDict({
        'friendly_name': 'Test Scheduled charging pending',
      }),
      'context': <ANY>,
      'entity_id': 'binary_sensor.test_scheduled_charging_pending',
      'last_changed': <ANY>,
      'last_updated': <ANY>,
      'state': 'off',
    }),
    StateSnapshot({
      'attributes': ReadOnlyDict({
        'friendly_name': 'Test Trip charging',
      }),
      'context': <ANY>,
      'entity_id': 'binary_sensor.test_trip_charging',
      'last_changed': <ANY>,
      'last_updated': <ANY>,
      'state': 'off',
    }),
    StateSnapshot({
      'attributes': ReadOnlyDict({
        'device_class': 'heat',
        'friendly_name': 'Test Auto seat climate left',
      }),
      'context': <ANY>,
      'entity_id': 'binary_sensor.test_auto_seat_climate_left',
      'last_changed': <ANY>,
      'last_updated': <ANY>,
      'state': 'on',
    }),
    StateSnapshot({
      'attributes': ReadOnlyDict({
        'device_class': 'heat',
        'friendly_name': 'Test Auto seat climate right',
      }),
      'context': <ANY>,
      'entity_id': 'binary_sensor.test_auto_seat_climate_right',
      'last_changed': <ANY>,
      'last_updated': <ANY>,
      'state': 'on',
    }),
    StateSnapshot({
      'attributes': ReadOnlyDict({
        'device_class': 'heat',
        'friendly_name': 'Test Heat',
      }),
      'context': <ANY>,
      'entity_id': 'binary_sensor.test_heat',
      'last_changed': <ANY>,
      'last_updated': <ANY>,
      'state': 'on',
    }),
    StateSnapshot({
      'attributes': ReadOnlyDict({
        'device_class': 'running',
        'friendly_name': 'Test Cabin overheat protection',
      }),
      'context': <ANY>,
      'entity_id': 'binary_sensor.test_cabin_overheat_protection',
      'last_changed': <ANY>,
      'last_updated': <ANY>,
      'state': 'on',
    }),
    StateSnapshot({
      'attributes': ReadOnlyDict({
        'device_class': 'heat',
        'friendly_name': 'Test Cabin overheat protection actively cooling',
      }),
      'context': <ANY>,
      'entity_id': 'binary_sensor.test_cabin_overheat_protection_actively_cooling',
      'last_changed': <ANY>,
      'last_updated': <ANY>,
      'state': 'off',
    }),
    StateSnapshot({
      'attributes': ReadOnlyDict({
        'device_class': 'running',
        'friendly_name': 'Test Dashcam',
      }),
      'context': <ANY>,
      'entity_id': 'binary_sensor.test_dashcam',
      'last_changed': <ANY>,
      'last_updated': <ANY>,
      'state': 'on',
    }),
    StateSnapshot({
      'attributes': ReadOnlyDict({
        'device_class': 'presence',
        'friendly_name': 'Test User present',
      }),
      'context': <ANY>,
      'entity_id': 'binary_sensor.test_user_present',
      'last_changed': <ANY>,
      'last_updated': <ANY>,
      'state': 'off',
    }),
    StateSnapshot({
      'attributes': ReadOnlyDict({
        'device_class': 'problem',
        'friendly_name': 'Test Tire pressure warning front left',
      }),
      'context': <ANY>,
      'entity_id': 'binary_sensor.test_tire_pressure_warning_front_left',
      'last_changed': <ANY>,
      'last_updated': <ANY>,
      'state': 'off',
    }),
    StateSnapshot({
      'attributes': ReadOnlyDict({
        'device_class': 'problem',
        'friendly_name': 'Test Tire pressure warning front right',
      }),
      'context': <ANY>,
      'entity_id': 'binary_sensor.test_tire_pressure_warning_front_right',
      'last_changed': <ANY>,
      'last_updated': <ANY>,
      'state': 'off',
    }),
    StateSnapshot({
      'attributes': ReadOnlyDict({
        'device_class': 'problem',
        'friendly_name': 'Test Tire pressure warning rear left',
      }),
      'context': <ANY>,
      'entity_id': 'binary_sensor.test_tire_pressure_warning_rear_left',
      'last_changed': <ANY>,
      'last_updated': <ANY>,
      'state': 'off',
    }),
    StateSnapshot({
      'attributes': ReadOnlyDict({
        'device_class': 'problem',
        'friendly_name': 'Test Tire pressure warning rear right',
      }),
      'context': <ANY>,
      'entity_id': 'binary_sensor.test_tire_pressure_warning_rear_right',
      'last_changed': <ANY>,
      'last_updated': <ANY>,
      'state': 'off',
    }),
  ])
# ---
